src/app/DomainPermissionsPage.qml

```diff
@@ -0,0 +1,290 @@
+                domainPermissionsListView.currentIndex = index;
+                return;
+            }
+        }
+    }
+
+    property QtObject domainPermissionsObject
+    property bool selectMode
+    property bool sortByLastRequested: false
+
+    signal done()
+    signal reload(string selectedDomain)
+
+    BrowserPage {
+        id: domainPermissionsPage
+
+        anchors.fill: parent
+        focus: true
+
+        title: i18n.tr("Domain specific permissions")
+
+        showBackAction: !selectMode
+
+        leadingActions: [
+            Action {
+                objectName: "close"
+                iconName: "close"
+                onTriggered: selectMode = false
+            }
+        ]
+
+        trailingActions: [
+            Action {
+                text: i18n.tr("Select all")
+                iconName: "select"
+                visible: selectMode
+                onTriggered: {
+                    if (domainPermissionsListView.ViewItems.selectedIndices.length === domainPermissionsListView.count) {
+                        domainPermissionsListView.ViewItems.selectedIndices = []
+                    } else {
+                        var indices = []
+                        for (var i = 0; i < domainPermissionsListView.count; ++i) {
+                            indices.push(i)
+                        }
+                        domainPermissionsListView.ViewItems.selectedIndices = indices
+                    }
+                }
+            },
+            Action {
+                text: i18n.tr("Delete")
+                iconName: "delete"
+                visible: selectMode
+                enabled: domainPermissionsListView.ViewItems.selectedIndices.length > 0
+                onTriggered: {
+                    var toDelete = []
+                    for (var index = 0; index < domainPermissionsListView.ViewItems.selectedIndices.length; index++) {
+                        var selectedDomainSetting = domainPermissionsListView.model.get(domainPermissionsListView.ViewItems.selectedIndices[index])
+                        toDelete.push(selectedDomainSetting.domain)
+                    }
+                    for (var i = 0; i < toDelete.length; i++) {
+                        DomainPermissionsModel.removeEntry(toDelete[i])
+                    }
+                    domainPermissionsListView.ViewItems.selectedIndices = []
+                    selectMode = false
+                }
+            },
+            Action {
+                iconName: "edit"
+                visible: !selectMode
+                enabled: domainPermissionsListView.count > 0
+                onTriggered: {
+                    selectMode = true
+                }
+            },
+            Action {
+                iconName: "add"
+                visible: !selectMode
+
+                onTriggered: {
+                    var promptDialog = PopupUtils.open(Qt.resolvedUrl("PromptDialog.qml"), domainPermissionsPage);
```

Comment:
Does `PromptDialog` allow giving the text box a URL type? It seems strange to have it listed as a normal text box. Also makes autocorrect go nuts.

Reply:
yeah that is a good point, I try adding that property to the prompt dialog.